### PR TITLE
Update relational-vs-nosql-data.md

### PR DESCRIPTION
**Pull request by azarboon**
## Summary

Firstly, I wonder why ACID has been only mentioned in this page but without any explanation or link for further read. I added a bit of information to disambiguate it a bit.

Secondly, "consistency" has different notions and means different things in ACID vs CAP: [in short](https://budibase.com/blog/data/cap-vs-acid)

> consistency in the context of CAP means that all nodes store and provide the same data. For ACID, consistency means that internal rules within an individual node must apply across the board. For example, that all values in a column must be unique.

So, I've added a clarification to inform readers and prevent confusion. 

Here you can find some books that backup this claim (You can Google to find more sources):

[Database Internals - A Deep Dive Into How Distributed Data Systems Work page 218 ](https://www.google.com.ph/books/edition/Database_Internals/-l2vDwAAQBAJ?hl=en&gbpv=1&dq=consistency%20in%20cap%20is%20different%20from%20acid&pg=PA218&printsec=frontcover)
[Big Data SMACK - A Guide to Apache Spark, Mesos, Akka, Cassandra, and Kafka page 213](https://www.google.com.ph/books/edition/Big_Data_SMACK/6-sqDQAAQBAJ?hl=en&gbpv=1&dq=consistency%20cap%20acid%20different&pg=PA213&printsec=frontcover)

Also, I noticed some inconsistencies for the terms used or the way they are written. Changed them a bit to make the page more consistent.

Fixes #Issue_Number (if available)

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/architecture/cloud-native/relational-vs-nosql-data.md](https://github.com/dotnet/docs/blob/53e9af500e61e110b4498ddb14160620083abc9a/docs/architecture/cloud-native/relational-vs-nosql-data.md) | [SQL vs. NoSQL data](https://review.learn.microsoft.com/en-us/dotnet/architecture/cloud-native/relational-vs-nosql-data?branch=pr-en-us-41641) |

### CODE REVIEW

**Comment by azarboon** [comment]
I'm using corporate laptop so kindly please help me with linting.

File: docs/architecture/cloud-native/relational-vs-nosql-data.md [hunk 1]
```diff
@@ -40,7 +41,7 @@ As a way to understand the differences between these types of databases, conside
 
 The theorem states that distributed data systems will offer a trade-off between consistency, availability, and partition tolerance. And, that any database can only guarantee *two* of the three properties:
 
-- *Consistency.* Every node in the cluster responds with the most recent data, even if the system must block the request until all replicas update. If you query a "consistent system" for an item that is currently updating, you'll wait for that response until all replicas successfully update. However, you'll receive the most current data.
+- *Consistency.* Every node in the cluster responds with the most recent data, even if the system must block the request until all replicas update. If you query a "consistent system" for an item that is currently updating, you'll wait for that response until all replicas successfully update. However, you'll receive the most current data. Note that "consistency" as defined by CAP theorem is technically different from "consistency" as defined in ACID guarantees.
```
Comment (IEvangelist) [inline docs/architecture/cloud-native/relational-vs-nosql-data.md:44]:
```suggestion
- *Consistency.* Every node in the cluster responds with the most recent data, even if the system must block the request until all replicas update. If you query a "consistent system" for an item that is currently updating, you'll wait for that response until all replicas successfully update. However, you'll receive the most current data. It should be understood that the term "consistency" as it's used in the context of the CAP theorem has a technical meaning that is distinct from the way "consistency" is defined in the context of ACID guarantees.
```

File: docs/architecture/cloud-native/relational-vs-nosql-data.md [hunk 2]
```diff
@@ -5,15 +5,16 @@ author: robvet
...
+*Relational databases* have been a prevalent technology for decades. They're mature, proven, and widely implemented. Competing database products, tooling, and expertise abound. Relational databases provide a store of related data tables. These tables have a fixed schema, use SQL (Structured Query Language) to manage data, and support ACID guarantees: atomicity, consistency, isolation and durability.
 
-*No-SQL databases* refer to high-performance, non-relational data stores. They excel in their ease-of-use, scalability, resilience, and availability characteristics. Instead of joining tables of normalized data, NoSQL stores unstructured or semi-structured data, often in key-value pairs or JSON documents. No-SQL databases typically don't provide ACID guarantees beyond the scope of a single database partition. High volume services that require sub second response time favor NoSQL datastores.
+
```
Comment (IEvangelist) [inline docs/architecture/cloud-native/relational-vs-nosql-data.md:16]:
```suggestion
```

**Review by IEvangelist — APPROVED** [review]
I like these additions, thank you @azarboon. I have a few minor suggestions for improvement to consider.

**Comment by azarboon** [comment]
@IEvangelist thanks. I accepted your suggestions.

**Pull request merged**